drgn/helpers/memory.py

```diff
@@ -0,0 +1,104 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+"""
+Memory
+------
+
+The ``drgn.helpers.memory`` module provides helpers for working with memory and addresses.
+"""
+
+import operator
+from typing import Optional, Union, overload
+
+import drgn
+from drgn import IntegerLike, Object, Program, SymbolKind
+from drgn.helpers import escape_ascii_string
+from drgn.helpers.linux.slab import find_containing_slab_cache
+
+__all__ = ("identify_address",)
+
+
+SYMBOL_KIND_STR = {
+    SymbolKind.UNKNOWN: "symbol",
+    SymbolKind.OBJECT: "symbol object",
+    SymbolKind.FUNC: "symbol function",
+    SymbolKind.SECTION: "symbol section",
+    SymbolKind.FILE: "symbol file",
+    SymbolKind.COMMON: "symbol common",
+    SymbolKind.TLS: "symbol tls",
+    SymbolKind.IFUNC: "symbol ifunc",
+}
+
+
+@overload
+def identify_address(addr: Object) -> Optional[str]:
+    """
+    Identify an address (given as an :class:`.Object`) as a symbol or slab object.
+
+    This will currently identify the following types of addresses as follows:
+
+    * Symbols (e.g., addresses in functions or global variables): `symbol symbol_kind symbol_name+offset` or `symbol symbol_kind symbol` if offset is 0.
+    * Slab objects: `slab object slab_cache_name`.
```

Comment:
This is inconsistent with the code, which has an additional `:`. I like the colon, so let's keep that and update the documentation. In fact, I think it'd be nice to add a colon for symbols, as well (i.e., `function symbol: foo`).

Reply:
Fixed.